bit_manipulation/find_non_repeating_number.cpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file
+ * @brief Implementation to find the non repeating integer
+ * in an array of repeating integers. [Single
+ * Number](https://leetcode.com/problems/single-number/)
+ *
+ * @details
+ * Given an array of integers in which all of the numbers occur exactly
+ * twice except one integer which occurs only once. Find the non-repeating
+ * integer.
+ *
+ * Worst Case Time Complexity: O(n)
+ * Space complexity: O(1)
+
+ * @author [Ravidev Pandey](https://github.com/literalEval)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+#include <vector>    /// storing the numbers
+
+/**
+ * @namespace bit_manipulation
+ * @brief Bit manipulation algorithms
+ */
+namespace bit_manipulation {
+/**
+ * @namespace setKthBit
+ * @brief Functions to find the non repeating integer
+ * in an array of repeating integers. [Single
+ * Number](https://leetcode.com/problems/single-number/)
+ */
+namespace find_single_number {
+/**
+ * @brief The main function implements find single number
+ * @param nums vector of integers
+ * @returns returns the integer that occurs only once
+ */
+int find_single_number(std::vector<int>& nums) {
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or on our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂